src/pyscipopt/scip.pyx

```diff
@@ -1543,6 +1547,46 @@ cdef class Model:
         PY_SCIP_CALL(SCIPgetTransformedCons(self._scip, cons.cons, &transcons))
         return Constraint.create(transcons)
 
+    def getBilintermsQuadratic(self, Constraint cons):
+        cdef SCIP_BILINTERM* _bilinterms
+        cdef int _nbilinterms
+
+        _bilinterms = SCIPgetBilinTermsQuadratic(self._scip, cons.cons)
+        _nbilinterms = SCIPgetNBilinTermsQuadratic(self._scip, cons.cons)
+
+        vars1 = [Variable.create(_bilinterms[i].var1) for i in range(_nbilinterms)]
+        vars2 = [Variable.create(_bilinterms[i].var2) for i in range(_nbilinterms)]
+        coefs = [_bilinterms[i].coef for i in range(_nbilinterms)]
+
+        return (vars1,vars2,coefs)
+
+    def getQuadtermsQuadratic(self, Constraint cons):
+        cdef SCIP_QUADVARTERM* _quadterms
+        cdef int _nquadterms
+
+        _quadterms = SCIPgetQuadVarTermsQuadratic(self._scip, cons.cons)
+        _nquadterms = SCIPgetNQuadVarTermsQuadratic(self._scip, cons.cons)
+
+        vars = [Variable.create(_quadterms[i].var) for i in range(_nquadterms)]
+        sqrcoefs = [_quadterms[i].sqrcoef for i in range(_nquadterms)]
+        lincoefs = [_quadterms[i].lincoef for i in range(_nquadterms)]
+
+        return (vars,sqrcoefs,lincoefs)
+
+    def getLintermsQuadratic(self, Constraint cons):
+        cdef SCIP_VAR** _linvars
+        cdef SCIP_Real* _lincoefs
+        cdef int _nlinvars
+
+        _linvars = SCIPgetLinearVarsQuadratic(self._scip, cons.cons)
+        _lincoefs = SCIPgetCoefsLinearVarsQuadratic(self._scip, cons.cons)
+        _nlinvars = SCIPgetNLinearVarsQuadratic(self._scip, cons.cons)
+
+        vars = [Variable.create(_linvars[i]) for i in range(_nlinvars)]
+        coefs = [_lincoefs[i] for i in range(_nlinvars)]
+
+        return (vars,coefs)
+
```

Comment:
In our use-case, we want to iterate through all quadratic constraints and just get all the corresponding non-zero values. In this case, mapping only the name of the variables isn't that useful because we need to have access to the variables.

What does it mean for this case? You want to return a list of variables and dictonaries to access the coefficients? This is unnecessary overhead I think we don't need to introduce here.

Reply:
It's actually less overhead in my code, because I'm only querying the names of the variables and use them as keys in the dict. I can then loop over all variable names in the problem and construct the constraint matrix from the dicts.

Reply:
Why not use `getVars()` to get all variables only once? Don't you possibly create multiple Python variables from the each variable in your example?

Reply:
How would you do it? You need to access the variables via `model.getVars()`, which then create the python variables. Also, would you really loop over all variables to just get a very few nonzeros for a constraint? This sounds terrible...

Reply:
It really depends on what you what you want to do with the data. In general, I prefer having connected data in connected data structures, e.g. dicts for varname, coeff. But yeah, I agree, the list that is returned by `getVars` is not easily hashable by variable name so you would indeed need to loop it.

And what about combining your 3 methods into one? Wouldn't this be more efficient, because you would usually want to have all coefficients, right?

Reply:
Okay, here is the following deal: I aggregate the three functions to a single one and don't return a triple/tuple of arrays but instead arrays of tuple/triple. The output for `x*y + x^2 + y^2` would be then for the bilinear terms `[(x,y,1)]` and for the quadratic ones `[(x,1),(y,1)]`.